File: app/src/main/java/com/arjanvlek/oxygenupdater/internal/Worker.java
package com.arjanvlek.oxygenupdater.internal;

import java8.lang.FunctionalInterface;

/**
 * Oxygen Updater - © 2017 Arjan Vlek
 */

@FunctionalInterface
public interface Worker {

	Worker NOOP = () -> {
	};

	void start();
}

File: app/src/main/java/com/arjanvlek/oxygenupdater/internal/Worker.kt
package com.arjanvlek.oxygenupdater.internal

import java8.lang.FunctionalInterface

/**
 * Oxygen Updater - © 2017 Arjan Vlek
 */

@FunctionalInterface
interface Worker {

    fun start()

    companion object {
        val NOOP = object : Worker {
            override fun start() {}
        }
    }
}
